hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/message/JsonMessage.java

```diff
@@ -1,19 +1,24 @@
 package pl.allegro.tech.hermes.frontend.publishing.message;
 
+import com.google.common.collect.ImmutableMap;
 import pl.allegro.tech.hermes.api.ContentType;
 
+import java.util.Map;
+
 public class JsonMessage implements Message {
 
     private final String id;
     private final byte[] data;
     private final long timestamp;
     private final String partitionKey;
+    private final Map<String, String> extraRequestHeaders;
```

Comment:
Probably it will not be needed if you apply my comment about the backup storage.